src/notifications/settings/initialization.ts

```diff
@@ -40,167 +30,119 @@ export const addDefaultNotificationGroupSettings = (override = false) => {
+  const globalSettings = getAllGlobalNotificationSettingsFromStorage();
+
+  if (isEmpty(currentSettings)) return;
+
+  InteractionManager.runAfterInteractions(() => {
+    publishAndSaveNotificationSettings({ globalSettings, walletSettings: currentSettings, skipPreSave: true });
+  });
 };
 
 /**
- * Adds fresh disabled settings for all wallets that didn't have settings
- * schedules subscribing to owned wallets that haven't been initialized yet
- * schedules removing settings for wallets that are no longer prersent but removal failed previously
+ * schedules subscribing to wallets that haven't been successfully subscribed to yet
+ * called from NotificationsHandler run on every cold start
  */
-export const initializeNotificationSettingsForAllAddressesAndCleanupSettingsForRemovedWallets = (addresses: AddressWithRelationship[]) => {
-  const initializationState = _prepareInitializationState();
-  // Set of wallet addresses we have in app (unrelated to notification settings entries)
-  const walletAddresses = new Set(addresses.map(addressWithRelationship => addressWithRelationship.address));
-  const removedWalletsThatWereNotUnsubscribedProperly: string[] = [...initializationState.alreadySaved.keys()].filter(
-    address => !walletAddresses.has(address)
-  );
+export const initializeNotificationSettingsForAllAddresses = (addresses: AddressWithRelationship[]) => {
+  const newSettings = createInitialSettingsForAllAddresses(addresses);
+  const globalSettings = getAllGlobalNotificationSettingsFromStorage();
 
-  const queue = _prepareSubscriptionQueueAndCreateInitialSettings(addresses, initializationState);
+  if (!newSettings) return;
 
   InteractionManager.runAfterInteractions(() => {
-    _processSubscriptionQueue(queue);
+    publishAndSaveNotificationSettings({ globalSettings, walletSettings: newSettings });
   });
-
-  if (removedWalletsThatWereNotUnsubscribedProperly.length) {
-    InteractionManager.runAfterInteractions(() => {
-      removedWalletsThatWereNotUnsubscribedProperly.forEach(address => {
-        removeNotificationSettingsForWallet(address);
-      });
-    });
-  }
 };
 
 /**
  * Adds fresh disabled settings for all wallets that didn't have settings
+ * called from createWallet and generateAccount flows
  */
 export const initializeNotificationSettingsForAddresses = (addresses: AddressWithRelationship[]) => {
-  const initializationState = _prepareInitializationState();
+  const proposedSettings = createInitialSettingsForNewlyAddedAddresses(addresses);
+  const globalSettings = getAllGlobalNotificationSettingsFromStorage();
+  InteractionManager.runAfterInteractions(() => {
+    publishAndSaveNotificationSettings({ globalSettings, walletSettings: proposedSettings });
+  });
+};
 
-  const queue = _prepareSubscriptionQueueAndCreateInitialSettings(addresses, initializationState);
+const createInitialSettingsForAllAddresses = (addresses: AddressWithRelationship[]) => {
```

Comment:
this function uses `needsUpdate` flag so that we only resubscribe when we need to (e.g, if `successfullyFinishedInitialSubscription` is set to `false` or if a wallet address that we have is missing from the existing notification settings)